auto_submit/test/src/service/fake_github_service.dart

```diff
@@ -7,21 +7,74 @@ import 'dart:convert';
 import 'package:auto_submit/service/github_service.dart';
 import 'package:github/github.dart';
 
-import '../../requests/github_webhook_test_data.dart';
 import '../../utilities/mocks.dart';
 
 /// A fake GithubService implementation.
 class FakeGithubService implements GithubService {
-  FakeGithubService({GitHub? client}) : github = client ?? MockGitHub();
+  FakeGithubService({
+    GitHub? client,
+    String? reviewsMock,
+    String? checkRunsMock,
+    String? repositoryStatusesMock,
+    String? commitMock,
+    String? compareTowCommitsMock,
+    String? successMergeMock,
+    String? createCommentMock,
+  }) : github = client ?? MockGitHub();
 
   @override
   final GitHub github;
 
+  String? reviewsMock;
+  String? checkRunsMock;
+  String? repositoryStatusesMock;
+  String? commitMock;
+  String? compareTowCommitsMock;
+  String? successMergeMock;
+  String? createCommentMock;
+
+  set reviewsData(String? reviewsMock) {
+    this.reviewsMock = reviewsMock;
+  }
+
+  set checkRunsData(String? checkRunsMock) {
+    this.checkRunsMock = checkRunsMock;
+  }
+
+  set repositoryStatusesData(String? repositoryStatusesMock) {
+    this.repositoryStatusesMock = repositoryStatusesMock;
+  }
+
+  set commitData(String? commitMock) {
+    this.commitMock = commitMock;
+  }
+
+  set compareTowCommitsData(String? compareTowCommitsMock) {
+    this.compareTowCommitsMock = compareTowCommitsMock;
+  }
+
+  set successMergeData(String? successMergeMock) {
+    this.successMergeMock = successMergeMock;
+  }
+
+  set createCommentData(String? createCommentMock) {
+    this.createCommentMock = createCommentMock;
+  }
+
   @override
   Future<List<PullRequestReview>> getReviews(RepositorySlug slug, int prNumber) async {
-    final List<dynamic> reviews = json.decode(reviewsMock) as List;
-    final List<PullRequestReview> prReviews =
-        reviews.map((dynamic review) => PullRequestReview.fromJson(review)).toList();
+    final List<Map<String, dynamic>> reviewsBody =
+        (jsonDecode(reviewsMock!) as List).map((e) => e as Map<String, dynamic>).toList();
+    final List<PullRequestReview> prReviews = <PullRequestReview>[];
+    for (Map reviewMap in reviewsBody) {
+      PullRequestReview review = PullRequestReview(
+          id: reviewMap['id'] as int,
+          body: reviewMap['body'] as String?,
+          state: reviewMap['state'] as String?,
+          user: User(id: reviewMap['user']['id'], login: reviewMap['user']['login']))
+        ..authorAssociation = reviewMap['author_association'] as String?;
+      prReviews.add(review);
```

Comment:
As we already simplify the mock data, we can switch back to `PullRequestReview.fromJson`? Same for other override functions.

Reply:
Sure, switched.